packages/core/src/commands/remove-node.ts

```diff
@@ -0,0 +1,63 @@
+import type { NodeType, ResolvedPos } from '@prosekit/pm/model'
+import type { Command } from '@prosekit/pm/state'
+
+import type { CommandCreator } from '../types/command'
+import { getNodeType } from '../utils/get-node-type'
+
+/**
+ * Returns a command that removes the given type of node at the current selection or at
+ * the given position.
+ *
+ * @public
+ */
+export function removeNode(options: {
+  /**
+   * The type of the mark to remove.
+   */
+  type: string | NodeType
+
+  /**
+   * The position of the document. By default it will be the anchor position of current selection.
+   */
+  position?: number
+}): Command {
+  return (state, dispatch) => {
+    const nodeType = getNodeType(state.schema, options.type)
+    const $pos =
+      typeof options.position === 'number'
+        ? state.doc.resolve(options.position)
+        : state.selection.$anchor
+
+    const getNodePos = (
+      $pos: ResolvedPos,
+    ): { from: number | null; to: number | null } => {
+      for (let depth = $pos.depth; depth > 0; depth -= 1) {
+        const node = $pos.node(depth)
+
+        if (node.type === nodeType) {
+          const from = $pos.before(depth)
+          const to = $pos.after(depth)
+          return {
+            from,
+            to,
+          }
+        }
+      }
```

Comment:
This function is only used once after defined. Could we inline it?

Reply:
Or better if we can add a `findParentNode()` function under `packages/core/src/utils/`.

Reply:
I've extracted the util.